include/bsp/xi_bsp_fwu.h

```diff
@@ -84,6 +84,28 @@ void xi_bsp_fwu_on_package_download_failure();
 
 /**
  * @function
+ * @brief Invoked at the start of a SFT download process, when SFT reports
+ * to the client that there are new resources to download.  This function
+ * lets the client application select the order in which resources will be
+ * downloaded.
+ *
+ * @param [in] resource_names an array of resource name strings that have
+ * new revisions in the SFT download package.
+ * @param [in] list_len the number of elements in the resource_names array.
+ * @param [in/out] download_order an integer array to be filled out by the
+ * BSP implementation.  The values in this array must be indicies of the
+ * resource_names array, ordered in the sequence that the resources should be
+ * downloaded. Values must be between 0 and list_len - 1.  When this function
+ * is invoked, this array is prepopulated with a default download order --
+ * this function may return immediately without altering the array to use
+ * these defaults.
```

Comment:
I'd add here a small example for explanation:
`For a list of 4 the default order is: [0, 1, 2, 3]. The application might alter this and return and array like this: [2, 3, 1, 0]`. This says a lot, at least for me, that this is not a rocket science, just reorder if you want.

Reply:
:+1: